### DISCUSSION TITLE
Why not use class to define types and initial state?

### DISCUSSION THREAD

**Discussion by AmitJoki** [vuejs/pinia#2160] [Help and Questions] [answered]
Is there a reason the [docs](https://pinia.vuejs.org/core-concepts/state.html) do not suggest the use of classes to define the state?

Here's the suggested way which duplicates the state structure twice, once for defining the types and then for the initial state.

```
interface State {
  userList: UserInfo[]
  user: UserInfo | null
}

export const useUserStore = defineStore('user', {
  state: (): State => {
    return {
      userList: [],
      user: null,
    }
  },
})

```

When we could do it all at once using classes which takes care of both in one go?

```
class State {
  userList: UserInfo[] = [];
  user: UserInfo | null = null;
}

export const useUserStore = defineStore('user', {
  state: () => new State()
})
```

**Comment by posva** [comment, marked as answer]
Classes are not serializable so this breaks when doing SSR.
